dts/msm8916-samsung-r06.dts

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+/include/ "msm8916.dtsi"
+
+/ {
+	// This is used by the bootloader to find the correct DTB
+	qcom,msm-id = <206 0>;
+	qcom,board-id = <0xCE08FF01 6>;
+
+	gt58ltebmc {
+		model = "Samsung Galaxy Tab A 8.0 (CAN)";
+		compatible = "samsung,gt58ltebmc", "qcom,msm8916-mtp", "qcom,msm8916", "qcom,mtp", "lk2nd,device";
```

Comment:
Like the `model`, this does not actually need to match what is used in downstream. Therefore you definitely don't need the `mtp` compatibles, because your tablet actually has nothing in common with MTP (a reference development board with MSM8916)

```suggestion
		compatible = "samsung,gt58ltebmc", "qcom,msm8916", "lk2nd,device";
```

Not sure about `samsung,gt58ltebmc`. I guess I would have called it `samsung,gt58lte-can` or `samsung,gt58lte-canbmc` like the downstream device tree (`msm8916-sec-gt58lte-canbmc-r06.dts`). `gt58ltebmc` is the Android codename as far as I can tell so that would be fine too.

We have no obligation to follow any particular name here, so it's up to you which one you would prefer using to refer to your device in the future. You can also keep it as is if you want.